examples/diagnostic_script.py

```diff
@@ -0,0 +1,195 @@
+        )
+    )
+    total_attachments_assets = attachment_identities_assets(assets_with_attachments)
+
+    # Total Number of Events
+    events = list(arch.events.list(props=props, attrs=attrs))
+
+    # Events with Extended Attributes
+    extended_attributes_event = events_ext_attr(events)
+
+    # Events with Attachments
+    events_with_attachments = list(
+        arch.events.list(
+            props=props,
+            attrs=dict(list(attrs.items()) + list({"arc_attachments": "*"}.items())),
+        )
+    )
+    total_attachments_events = attachment_identities_events(events_with_attachments)
+
+    # Average Events per Asset
+    avg_events_per_asset = len(events) / len(assets)
+
+    # Total Number of Subjects
+    subjects = list(arch.subjects.list())
+
+    # Total Number of Access Policies
+    access_policies = list(arch.access_policies.list())
+
+    summary_output = {
+        "Date of Scan": str(datetime.now()),
+        "Number of Assets": len(assets),
+        "Assets with Extended Attributes": len(extended_attributes_asset),
+        "Assets with Associated Location": len(assets_with_location),
+        "Number of Events": len(events),
+        "Events with Extended Attributes": len(extended_attributes_event),
+        "Events with Attachments": len(events_with_attachments),
+        "Average Events per Asset": round(avg_events_per_asset),
+        "Total Number of Attachments": len(total_attachments_assets)
+        + len(total_attachments_events),
+        "Number of Subjects": len(subjects),
+        "Number of Access Policies": len(access_policies),
+    }
+
+    # Create directories for output storage
+    directory_a = "total_estate"
+    directory_b = "assets_with_associated_events"
+    parent_directory = os.path.abspath(os.getcwd())
+    path_a = os.path.join(parent_directory, directory_a)
+    path_b = os.path.join(path_a, directory_b)
+    os.makedirs(path_a, exist_ok=True)
+    os.makedirs(path_b, exist_ok=True)
+
+    # Create a zip file containing detials of each asset, event, etc.
+    estate_list = [assets, events, subjects, access_policies, summary_output]
+    file_list = ["assets", "events", "subjects", "access_policies", "summary"]
+    for item, name in zip(estate_list, file_list):
+        with open(f"{name}.json", "w", encoding="utf8") as outfile:
+            json.dump(item, outfile, indent=4)
+        if os.path.exists(os.path.join(path_a, f"{name}.json")):
+            os.remove(os.path.join(path_a, f"{name}.json"))
+        shutil.move(os.path.join(parent_directory, f"{name}.json"), path_a)
+
+    # Create folder with files for each asset and associated events.
+    list_groups = []
+    for asset in assets:
+        event_list = []
+        for event in events:
+            if asset["identity"] == event["asset_identity"]:
+                event_list.append(event)
+        list_groups.append({str(asset["identity"]): event_list})
+
```

Comment:
event_list = [e for e in events if asset["identity"] == e["identity"]]

replaces 4 lines

Reply:
change made